cmake/add_enclave.cmake

```diff
@@ -121,15 +122,23 @@ endfunction()
 
 function(add_enclave_sgx)
   set(options CXX)
-  set(oneValueArgs TARGET CONFIG KEY SIGNING_ENGINE ENGINE_LOAD_PATH ENGINE_KEY_ID)
+  set(oneValueArgs TARGET CONFIG KEY SIGNING_ENGINE ENGINE_LOAD_PATH ENGINE_KEY_ID ENABLE_LVI_MITIGATION)
   set(multiValueArgs SOURCES)
   cmake_parse_arguments(ENCLAVE
     "${options}"
     "${oneValueArgs}"
     "${multiValueArgs}"
     ${ARGN})
 
-  add_enclave_executable(${ENCLAVE_TARGET} ${ENCLAVE_SOURCES})
+  # Add an enclave with LVI mitigation when the `ENABLE_LVI_MITIGATION` argument or the `ENABLE_LVI_MITIGATION_TESTS`
+  # global variable is explicitly set.
+  if (LVI_MITIGATION MATCHES ControlFlow AND
```

Comment:
Should this logic be in `add_enclave_executable` instead?

Reply:
It then requires additional layer of parameter passing (i.e., the `ENABLE_LVI_MITIGATION`) to `add_enclave_executable`. As a result, I would still prefer to put the logic here (just extending the argument list of `add_enclave`).

The reason for having this parameter is to allow building certain tests (currently, mbedtls, libc, and libcxx) against LVI-mitigated libraries (but not run) by default when building with `LVI_MITIGATION=ControlFlow`. So that the CI build will still test linking those libraries.

Any other suggestion for doing this is welcome.

@CodeMonkeyLeet Please also take a look.

Reply:
It seems a little confusing that `ENABLE_LVI_MITIGATION_TESTS` only builds the tests, they won't actually run with ctest, right?

I see what you're trying to do, I just worry that the logic is confusing.

Reply:
`ENABLE_LVI_MITIGATION_TESTS` is a global variable set during cmake initialization.
`ENABLE_LVI_MITIGATION_TESTS=ON` does both (build can run with ctest), while
`ENABLE_LVI_MITIGATION_TESTS=OFF` (default value) does nothing.

`ENABLE_LVI_MITIGATION` here is an argument passing into `add_enclave`, which results in building the enclave with LVI mitigation when it's set to `ON`. This is used to complement the default case so that we can selectively build some tests (but not run).

Reply:
Oh ok, I misunderstood. Thanks for clarifying